imap_processing/tests/spice/test_kernels.py

```diff
@@ -61,9 +93,65 @@ def test_ensure_spice_time_kernels():
 
 def test_ensure_spice_key_error():
     """Test functionality of ensure spice when all branches fail"""
-    wrapped = kernels.ensure_spice(spice.et2utc)
+    wrapped = ensure_spice(spice.et2utc)
     # The ensure_spice decorator should raise a SpiceyError when all attempts to
     # furnish a set of kernels with sufficient coverage for the spiceypy
     # functions that it decorates.
     with pytest.raises(SpiceyError):
         _ = wrapped(577365941.184, "ISOC", 3) == "2018-04-18T23:24:31.998"
+
+
+def test_average_quaternions(et_times, kernels):
+    """Tests average_quaternions function."""
+    # TODO: remove spice.furnsh(kernels) after ensure_spice update.
+    spice.furnsh(kernels)
+    q_avg = _average_quaternions(et_times)
+
+    # Generated from MATLAB code results
+    q_avg_expected = np.array([-0.6611, 0.4981, -0.5019, -0.2509])
+    np.testing.assert_allclose(q_avg, q_avg_expected, atol=1e-4)
+
+
+def test_create_rotation_matrix(et_times, kernels):
+    """Tests create_rotation_matrix function."""
+    # TODO: remove spice.furnsh(kernels) after ensure_spice update.
+    spice.furnsh(kernels)
+    rotation_matrix = _create_rotation_matrix(et_times)
+    q_avg = _average_quaternions(et_times)
+    z_avg = spice.q2m(list(q_avg))[:, 2]
+
+    rotation_matrix_expected = np.array(
+        [[0.0000, 0.0000, 1.0000], [0.9104, -0.4136, 0.0000], [0.4136, 0.9104, 0.0000]]
+    )
+    z_avg_expected = np.array([0.4136, 0.9104, 0.0000])
+
+    np.testing.assert_allclose(z_avg, z_avg_expected, atol=1e-4)
+    np.testing.assert_allclose(rotation_matrix, rotation_matrix_expected, atol=1e-4)
+
+
+def test_create_pointing_frame(spice_test_data_path, kernels, tmp_path):
+    """Tests create_pointing_frame function."""
+    # TODO: remove spice.furnsh(kernels) after ensure_spice update.
```

Comment:
Remove this TODO. This is a good way to test this code independent of the ensure_spice decorator.